### PR TITLE
Bypass v14

### PR DESCRIPTION
**Pull request by regit**
Addressing comments in #2293 

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/202
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/198

### CODE REVIEW

File: src/flow.c [hunk 1]
```diff
@@ -223,6 +223,19 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
...
 
+    int state = SC_ATOMIC_GET(f->flow_state);
+
+    if (state != FLOW_STATE_CAPTURE_BYPASSED) {
+        /* update the last seen timestamp of this flow */
+        COPY_TIMESTAMP(&p->ts, &f->lastts);
+    } else {
+        /* still seeing packet, we downgrade to local bypass */
+        if (p->ts.tv_sec - f->lastts.tv_sec > FLOW_BYPASSED_TIMEOUT / 2) {
+            COPY_TIMESTAMP(&p->ts, &f->lastts);
```
Comment (inliniac) [inline src/flow.c]: I think it could be interesting to do a counter for this case.

File: suricata.yaml.in [hunk 2]
```diff
@@ -1055,7 +1055,8 @@ vlan:
 # of that flow). The value of "established" is the amount of
 # seconds that the engine will wait to free the flow if it spend that amount
 # without receiving new packets or closing the connection. "closed" is the
-# amount of time to wait after a flow is closed (usually zero).
+# amount of time to wait after a flow is closed (usually zero). "bypassed" is
```
Comment (inliniac) [inline suricata.yaml.in]: Not sure this helps. I think it should explain wait the purpose is. Here one just wonders: if equivalent to established why does it exist?

Comment (inliniac) [inline suricata.yaml.in]: "bypass timeout controls locally bypassed flows. For these flows we don't do any other tracking. If no packets have been seen after this timeout, the flow is discarded".

Something like this.

File: src/flow.c (hunk 1 above)
Comment (regit) [inline src/flow.c]: Do you want the counter in this PR or can we do it later ?

File: suricata.yaml.in (hunk 2 above)
Comment (regit) [inline suricata.yaml.in]: ok

**Comment by inliniac** [comment]
On 27-09-16 15:30, Eric Leblond wrote:
> *@regit* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In src/flow.c <https://github.com/inliniac/suricata/pull/2294>:
> 
>> @@ -223,6 +223,19 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
>  {
>      SCLogDebug("packet %"PRIu64" -- flow %p", p->pcap_cnt, f);
>  
> +    int state = SC_ATOMIC_GET(f->flow_state);
> +
> +    if (state != FLOW_STATE_CAPTURE_BYPASSED) {
> +        /* update the last seen timestamp of this flow */
> +        COPY_TIMESTAMP(&p->ts, &f->lastts);
> +    } else {
> +        /* still seeing packet, we downgrade to local bypass */
> +        if (p->ts.tv_sec - f->lastts.tv_sec > FLOW_BYPASSED_TIMEOUT / 2) {
> +            COPY_TIMESTAMP(&p->ts, &f->lastts);
> 
> Do you want the counter in this PR or can we do it later ?

I can be done later, but if you're doing another round anyway... lets
just add it ;)

File: src/flow.c (hunk 1 above)
Comment (regit) [inline src/flow.c]: We don't have any tv* in the function. Do we find a path to push one there ?

**Comment by inliniac** [comment]
On 27-09-16 15:35, Eric Leblond wrote:
> *@regit* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In src/flow.c <https://github.com/inliniac/suricata/pull/2294>:
> 
>> @@ -223,6 +223,19 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
>  {
>      SCLogDebug("packet %"PRIu64" -- flow %p", p->pcap_cnt, f);
>  
> +    int state = SC_ATOMIC_GET(f->flow_state);
> +
> +    if (state != FLOW_STATE_CAPTURE_BYPASSED) {
> +        /* update the last seen timestamp of this flow */
> +        COPY_TIMESTAMP(&p->ts, &f->lastts);
> +    } else {
> +        /* still seeing packet, we downgrade to local bypass */
> +        if (p->ts.tv_sec - f->lastts.tv_sec > FLOW_BYPASSED_TIMEOUT / 2) {
> +            COPY_TIMESTAMP(&p->ts, &f->lastts);
> 
> We don't have any tv* in the function. Do we find a path to push one there ?

Hmmm I prefer to keep it out of there... needs some thought.

**Pull request closed without merging**